lib/ocpp/v201/functional_blocks/security.cpp

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+
+#include <ocpp/v201/functional_blocks/security.hpp>
+
+#include <ocpp/v201/ctrlr_component_variables.hpp>
+#include <ocpp/v201/messages/SecurityEventNotification.hpp>
+#include <ocpp/v201/utils.hpp>
+
+constexpr int32_t minimum_cert_signing_wait_time_seconds = 250;
+
+namespace ocpp::v201 {
+
+Security::Security(MessageDispatcherInterface<MessageType>& message_dispatcher, DeviceModel& device_model,
+                   MessageLogging& logging, EvseSecurity& evse_security,
+                   ConnectivityManagerInterface& connectivity_manager, OcspUpdaterInterface& ocsp_updater,
+                   SecurityEventCallback security_event_callback) :
+    message_dispatcher(message_dispatcher),
+    device_model(device_model),
+    logging(logging),
+    evse_security(evse_security),
+    connectivity_manager(connectivity_manager),
+    ocsp_updater(ocsp_updater),
+    security_event_callback(security_event_callback),
+    csr_attempt(1) {
+}
+
+Security::~Security() {
+    stop();
+}
+
+void Security::handle_message(const EnhancedMessage<MessageType>& message) {
+    const auto& json_message = message.message;
+    switch (message.messageType) {
+    case MessageType::CertificateSigned:
+        this->handle_certificate_signed_req(json_message);
+        break;
+    case MessageType::SignCertificateResponse:
+        this->handle_sign_certificate_response(json_message);
+        break;
+    default:
+        throw MessageTypeNotImplementedException(message.messageType);
+    }
+}
+
+void Security::stop() {
```

Comment:
I would simply name this function it to exactly what is does `stop_certificate_signed_timer()`. If more code is added to it, we may want to rename it to something more generic, but thats currently not the case.